mixer/pkg/il/interpreter/result.go

```diff
@@ -111,13 +111,10 @@ func (r Result) AsInterface() interface{} {
 	case il.Interface:
 		// TODO: currently byte[] can only be net.IP address type, so we can
 		// safely do this. Ideally we want to have a better fix.
-		if r.vi != nil {
-			switch r.vi.(type) {
-			case []byte:
-				return net.IP(r.vi.([]uint8))
-			}
+		if attribute.CheckType(r.vi) {
+			return r.vi
```

Comment:
I don't think this is right. Expressions should operate on the logical attribute types, not on the transport types.

Reply:
Expressions are compiled into IL which erases certain types (e.g. `Interface` type is a big bucket of everything-else). As IL interpreter executes, the inner interface instances are loosely typed from then on. This check is to ensure that the stuff in the result (as produced by foreign functions, for example), is well-typed.